### PR TITLE
py3: use format_exc instead of print_exception

### PR DESCRIPTION
**Pull request by mroutis**
Related: #1818

### CODE REVIEW

**Comment by Suor** [comment]
@mroutis you things like `cause_tb` here, no testing how `from exc` and implicit chaining works. Why did you close that? It was almost done.

File: dvc/logger.py [hunk 1]
```diff
@@ -101,14 +101,8 @@ def _description(self, message, exception):
...
     def _walk_exc(self, exc_info):
-        import traceback
-
-        buffer = io.StringIO()
-
-        traceback.print_exception(*exc_info, file=buffer)
-
         exc = exc_info[1]
-        tb = buffer.getvalue()
+        tb = traceback.format_exc(*exc_info)
```
Comment (efiop) [inline dvc/logger.py]: Note that we no longer need `cause_tb` down below, so `tb_list` will only contain one entry, so we can get rid of it.

**Comment by mroutis** [comment]
> @mroutis you left things like cause_tb here, no testing how from exc and implicit chaining works. Why did you close that? It was almost done.

It was too big and it addressed changes that were not related to python 3

File: dvc/logger.py [hunk 2]
```diff
@@ -102,27 +102,23 @@ def _description(self, message, exception):
 
     def _walk_exc(self, exc_info):
         exc = exc_info[1]
-        tb = traceback.format_exc(*exc_info)
 
         exc_list = [str(exc)]
-        tb_list = [tb]
 
-        # NOTE: parsing chained exceptions. See dvc/exceptions.py for more info
         while hasattr(exc, "__cause__") and exc.__cause__:
```
Comment (Suor) [inline dvc/logger.py]: Again, what about `__context__`?

Comment (mroutis) [reply dvc/logger.py]: @Suor , `__context__`? We are not using `__context__`, and that should be the matter of another PR, I guess

**Comment by mroutis** [comment]
> no testing how from exc and implicit chaining works

@Suor , implicit errors don't work with the current implementation.

File: dvc/logger.py (hunk 2 above)
Comment (Suor) [reply dvc/logger.py]: Python 3 automatically assigns `__context__` when you raise an exception while handling exception. So if you don't use either `from exc` or `from None` you still can show the original cause, this is usually an error in an error handler, which we probably want to see.

**Comment by Suor** [comment]
@mroutis implicit errors work when you `raise exc`, so if we are not showing them it means we broke it.

**Comment by efiop** [comment]
@Suor we didn't have `__context__` earlier and didn't have implicit exceptions, nor do we use them in our code (3rd party libs might, be we haven't run into that yet). So no need to address those in this PR.

**Pull request merged**